src/Kernel32.Tests/storebanned/Kernel32Facts.cs

```diff
@@ -944,6 +944,184 @@ public void SetHandleInformation_DoesNotThrow()
+    /// of that of the calling thread.
+    /// </summary>
+    [Fact]
+    public unsafe void CreateRemoteThread_PseudoTest()
+    {
+        var secAttrs = new Kernel32.SECURITY_ATTRIBUTES
+        {
+            bInheritHandle = 1,
+            lpSecurityDescriptor = IntPtr.Zero,
+            nLength = Marshal.SizeOf<Kernel32.SECURITY_ATTRIBUTES>()
+        };
+
+        var dwThreadId = Kernel32.GetCurrentThreadId();
+        using var hProcess = Kernel32.GetCurrentProcess();
+
+        var result = false;
+        var gcHandle = GCHandle.Alloc(result);
+        try
+        {
+            var dwNewThreadId = 0;
+            using var hThread =
+                    Kernel32.CreateRemoteThread(
+                        hProcess.DangerousGetHandle(),
+                        &secAttrs,
+                        SIZE_T.Zero,
+                        new Kernel32.THREAD_START_ROUTINE(CreateThread_Test_ThreadMain),
+                        GCHandle.ToIntPtr(gcHandle),
+                        Kernel32.CreateProcessFlags.None,
+                        &dwNewThreadId);
+            Kernel32.WaitForSingleObject(hThread, -1);
+
+            result = (bool)gcHandle.Target;
+            Assert.True(result);
+            Assert.NotEqual(dwThreadId, dwNewThreadId);
+        }
+        finally
+        {
+            gcHandle.Free();
+        }
+    }
+
+    /// <summary>
+    /// Basic validation for <see cref="CreateRemoteThreadEx(IntPtr, SECURITY_ATTRIBUTES*, SIZE_T, THREAD_START_ROUTINE, IntPtr, CreateProcessFlags, PROC_THREAD_ATTRIBUTE_LIST*, int*)"/>
+    /// Note that this test DOES NOT create a true REMOTE thread in a foreign process; it just leverages this function to create a thread in the current (i.e, the test) procrss.
+    /// Nevertheless, this approach provides modest confidence that the P/Invoke definition is well-formed.
+    ///
+    /// Creates a thread by supplying <see cref="CreateThread_Test_ThreadMain(IntPtr)"/> as its ThreadProc/<see cref="Kernel32.THREAD_START_ROUTINE"/>.
+    /// The ThreadProc updates a bool value (supplied by the thread that created it) from false -> true. This change
+    /// is observed by the calling thread as proof of successful thread-creation.
+    ///
+    /// Also validates that the (native) Thread-ID for the newly created Thread is different than the (native) Thread-ID
+    /// of that of the calling thread.
+    /// </summary>
+    [Fact]
+    public unsafe void CreateRemoteThreadEx_PseudoTest()
+    {
+        var secAttrs = new Kernel32.SECURITY_ATTRIBUTES
+        {
+            bInheritHandle = 1,
+            lpSecurityDescriptor = IntPtr.Zero,
+            nLength = Marshal.SizeOf<Kernel32.SECURITY_ATTRIBUTES>()
+        };
+
+        var dwThreadId = Kernel32.GetCurrentThreadId();
+        using var hProcess = Kernel32.GetCurrentProcess();
+
+        var result = false;
+        var gcHandle = GCHandle.Alloc(result);
```

Comment:
Given `result` is a local variable, I wonder why you need to allocate a GC handle for it. Isn't its direct memory address sufficient? The runtime never rearranges memory on the stack, right? 
As written this method will not pop off the stack before the new thread exits, but even if it did, GCHandle can't guarantee this location on the stack will remain, right?

Reply:
OK, looking at this more I guess it's implicitly boxing the `bool`, which is subtle, to say the least. Would it be more clear if you defined a `StrongBox<bool>` local variable and pinned that instead? Then you could access the `bool` any time instead of this odd creation of a local variable which you never use except to assign it later from the thing that *seemed* to pin it.